vllm/worker/enc_dec_model_runner.py

```diff
@@ -0,0 +1,478 @@
+                # tokens.
+                tokens = [encoder_seq_data.get_last_token_id()]
+
+                if cross_block_table is not None:
+                    # Decode
+                    block_table = cross_block_table
+
+            block_table = (
+                []  # Memory profiling
+                if
+                (is_prompt or cross_block_table is None) else cross_block_table
+            )  # Decode
+
+            cross_block_tables.append(block_table)
+
+            encoder_seq_lens.append(seq_len)
+            context_lens.append(context_len)
+            query_len = seq_len - context_len
+            query_lens.append(query_len)
+            encoder_input_tokens.extend(tokens)
+            encoder_input_positions.extend(list(range(context_len, seq_len)))
+            prompt_adapter_id = seq_group_metadata.prompt_adapter_id
+
+            if prompt_adapter_id > 0 and is_prompt:
+                prompt_adapter_requests.add(
+                    seq_group_metadata.prompt_adapter_request)
+
+                num_tokens = seq_group_metadata.\
+                                        prompt_adapter_num_virtual_tokens
+                pm = [prompt_adapter_id
+                      ] * num_tokens + [0] * (query_len - num_tokens)
+                prompt_adapter_index_mapping += pm
+                prompt_adapter_prompt_mapping.extend(
+                    [prompt_adapter_id] *
+                    (query_len if seq_group_metadata.sampling_params
+                     and seq_group_metadata.sampling_params.prompt_logprobs
+                     else 1))
+
+            is_profile_run = _is_single_block_table_empty(
+                seq_group_metadata.block_tables)
+            if is_profile_run:
+                # During memory profiling, the block tables are not
+                # initialized yet. In this case, we just use a dummy
+                # slot mapping.
+                # In embeddings, the block tables are {seq_id: None}.
+                cross_slot_mapping.extend([_PAD_SLOT_ID] * seq_len)
+                continue
+
+            # Compute the slot mapping.
+            block_table = cross_block_table
+
+            for i in range(context_len, seq_len):
+
+                block_number = block_table[i // self.block_size]
+                block_offset = i % self.block_size
+                slot = block_number * self.block_size + block_offset
+                cross_slot_mapping.append(slot)
+
+        max_query_len = max(query_lens)
+        encoder_max_seq_len = max(encoder_seq_lens, default=0)
+
+        max_cross_block_table_len = max(
+            len(block_table) for block_table in cross_block_tables)
+        cross_block_tables = make_tensor_with_pad(
+            cross_block_tables,
+            max_len=max_cross_block_table_len,
+            pad=0,
+            dtype=torch.int,
+            device=self.device,
+        )
+        assert (not is_prompt) or max_query_len > 0, (
```

Comment:
What are we trying to assert here? If is_prompt is false (decoder phase) then this assert will always be true and we will not verify the max_query_len. 

Are we allowing empty encoder prompts as input because this assert might fail if the user provides  empty encoder prompt/s.

Reply:
This is fair, it probably is not valuable to keep this check so I removed the whole assert statement.